### ISSUE TITLE
How can we train using both the train and validation datasets to prepare for test dataset submission?

### ISSUE THREAD

**Issue by qifeng22** [happinesslz/LION#26]
Thank you for your excellent work. Could you share the method for training on the entire dataset (train and validation) for test dataset submission in Waymo and nuScenes?
@happinesslz

**Comment by AlmoonYsl** [comment]
> Thank you for your excellent work. Could you share the method for training on the entire dataset (train and validation) for test dataset submission in Waymo and nuScenes? @happinesslz

Hi, you should combine training and validation set for generating gt database and info.

**Comment by qifeng22** [comment]
The meaning is that I should move the validation folder to train folder， and then run this command:
```
python -m pcdet.datasets.waymo.waymo_dataset --func create_waymo_infos \
    --cfg_file tools/cfgs/dataset_configs/waymo_dataset.yaml
```
or 
```
python -m pcdet.datasets.nuscenes.nuscenes_dataset --func create_nuscenes_infos \
    --cfg_file tools/cfgs/dataset_configs/nuscenes_dataset.yaml \
    --version v1.0-trainval
```

**Comment by qifeng22** [comment]
If I train all the dataset, how to generate the test result for the evaluation website?

**Comment by AlmoonYsl** [comment]
> If I train all the dataset, how to generate the test result for the evaluation website?

For waymo, you need to inference the test set and convert the prediction to binary file like gt.bin.
For nuScenes, you need to inference the test set and generate the json file.

**Comment by qifeng22** [comment]
Can you provide the command? Maybe I dont konw  how to do?

**Comment by AlmoonYsl** [comment]
> Can you provide the command? Maybe I dont konw how to do?

For Waymo, we develop the generation tool for converting the prediction to binary file. But we do not clean the code. You can refer to the MMDetection3D codebase.
For nuScenes, you can directly eval the model by the shell script for generating the json file in the out directory.

**Comment by qifeng22** [comment]
okok, thanks

**Comment by qifeng22** [comment]
How can I merge the train and val NuScenes datasets to use the entire dataset for training the model? I tried modifying the code in pcdet/datasets/nuscenes_dataset.py at line 263 by setting train_scenes = splits.train + splits.val, but I was unable to train on the merged dataset.
@AlmoonYsl
